exercises/simple-cipher/canonical-data.json

```diff
@@ -0,0 +1,145 @@
+{
+  "exercise": "simple-cipher",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Random key cipher",
+      "cases": [
+        {
+          "description": "Key is made only of lowercase letters",
+          "property": "key",
+          "input": {},
+          "expected": {
+            "match": "^[a-z]+$"
+          }
+        },
+        {
+          "description": "It can encode",
+          "property": "encode",
+          "input": {
+            "plaintext": "aaaaaaaaaa"
+          },
+          "expected": "cipher.key"
+        },
+        {
+          "description": "It can decode",
+          "property": "decode",
+          "input": {
+            "ciphertext": "cipher.key"
+          },
+          "expected": "aaaaaaaaaa"
+        },
+        {
+          "description": "It is reversible. I.e., if you apply decode in a encoded result, you must see the same plaintext encode parameter as a result of the decode method",
```

Comment:
This test is useful to show the student what is possible. However, it is not possible for the test to reject any implementation that the previous two tests have not already rejected, since it just composes the previous two tests together.

Thus, the question is: Is the advantage of showing the student the possibilities worth the disadvantage of the test that does not help catch any more mistakes?

I personally thought it was not worth keeping this test. Let me know your thoughts on it.

Reply:
The first test guarantee that if we pass `aaaaaaaaaa` as argument to the `encode`, the result should be the same as the key. And the second one guarantee that if we pass `key` as argument to the `decode` we expect a `aaaaaaaaaa` as result. This is the first test that uses a different `plaintext`.

I believe that is good to explicitly verify that the algorithm is reversible, since is a characteristic of it. But you're right, if the encode and the decode are already verified with valid `"reversibles"` outputs, the nesting of the encode and decode must be reversible.

I personally like this reversibles test cases. But I see your point